airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -294,6 +294,29 @@ public void succeedAttempt(long jobId, int attemptNumber) throws IOException {
     });
   }
 
+  @Override
+  public void setAttemptTemporalWorkflowId(long jobId, int attemptNumber, String temporalWorkflowId) throws IOException {
+    database.query(ctx -> ctx.execute(
+        " UPDATE attempts SET temporal_workflow_id = ? WHERE job_id = ? AND attempt_number = ?",
+        temporalWorkflowId,
+        jobId,
+        attemptNumber));
+  }
+
+  @Override
+  public String getAttemptTemporalWorkflowId(long jobId, int attemptNumber) throws IOException {
```

Comment:
Agree that migrating to an optional field is safer.

Reply:
Charles, I actually want to defer to you or @jrhizor  on this.

I went with the RTE because the existing code doesn't return an optional. `getJob` unwraps the optional and throws an RTE if the job isn't present.

From my experience, there is a lag between Temporal picking up the job, and the attempt being created. Since the workflow updates the workflow id, it is possible for an attempt to exist without a workflow id.

Reply:
Should we wait here for Temporal to pick up the job? Doing an `Optional` makes this just a "best effort" method of getting the workflow id, right?

Reply:
I think not waiting is okay.

I thought about this and Optional does seem better, since we might retrieve this before this is created and throwing/catching an exception is awkward.